### DISCUSSION TITLE
Integrating EDS enrichment map into EBSD dataset in MTEX

### DISCUSSION THREAD

**Discussion by Remsdel** [mtex-toolbox/mtex#1069] [Ask Anything] [answered]
Hello,
I have an EDS enrichment elemental map with a one-to-one pixel spatial correspondence with the EBSD map (basically a matrix of the same size than the EBSD map containing elemental weight percent values). I would like to embed my pixel enrichment data into the EBSD dataset in MTEX in order to extract enrichment value in function of EBSD parameters, for example the grain average elemental enrichment. What would be the most efficient way to integrate this data into the EBSD dataset?
Many thanks!
Rémi

**Comment by zmichels** [comment, marked as answer]
Hi There,

I have done this with CL data/images recently.

Lets say you have a matrix of EDS data in a variable called “EDS”, and your EBSD data in a variable named “ebsd."

I found it easiest to “gridify” the EBSD data for assigning a value from a matrix as a property to the EBSD variable.

So, maybe try something like this to see if it works…

%% gridify the ebsd dataset
ebsd = ebsd.gridify;
 
%% assign the (same-size) EDS data as a property of EBSD
ebsd.prop.EDS = EDS;
 
%% compute grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',[10 1]*degree,'boundary','tight');
 
%% compute a grain-averaged EDS value of some kind...
gEDS = accumarray(ebsd.grainId, ebsd.EDS, size(grains), @nanmean);




Cheers,
Zach

**Reply by Remsdel** [reply]
Thanks, works like a charm,
Cheers,
Rémi
